build.gradle.kts

```diff
@@ -500,3 +501,27 @@ tasks {
 }
 
 apply(plugin = "com.dorongold.task-tree")
+
+tasks.register("checkJDKVersionOnMacOsX") {
+  doLast {
+    if (OperatingSystem.current().isMacOsX()) {
+      val vendor = System.getProperty("java.vendor").lowercase(Locale.getDefault())
+      val jdkRuntime = System.getProperty("java.runtime.name")
+      println("Java vendor: $vendor, JDK runtime: $jdkRuntime")
+      if (JavaVersion.current() == JavaVersion.VERSION_17 && jdkRuntime.contains("OpenJDK") &&
+        (vendor.contains("oracle") || vendor.contains("homebrew"))
+      ) {
+        throw GradleException(
+          "Gravitino current doesn't support building with " +
+            "Java version: OpenJDK 17 on MacOsX. Please use other JDK 17."
+        )
+      }
+    }
+  }
+}
```

Comment:
I think only checking the jdk vendor is not enough, because gradle javatoolchain has its own mechanism to discover and use the JDK version. So instead of just checking the JDK version, you'd better also add some restrictions to java version to use our tested version or something else.

Reply:
```
tasks.all {
  if (this.name != "checkJDKVersionOnMacOsX") {
    dependsOn("checkJDKVersionOnMacOsX")
  }
}
```
The code above states that **all tasks** will check the version of the JDK they are using.

> I think only checking the jdk vendor is not enough, because gradle javatoolchain has its own mechanism to discover and use the JDK version

I noticed that the version of JDK that triggers gradle and gradle tasks may be different. Besides OpenJDK17, We haven't encountered any issues with other JDK 17 release versions, so I would like to exclude it first. 

> I think only checking the jdk vendor is not enough

Vendor and runtime information are as follows:
```text
Java vendor: amazon.com inc., JDK runtime: OpenJDK Runtime Environment
```

I think it's clear enough. Am I missing anything?